kore/src/Kore/Exec.hs

```diff
@@ -644,11 +651,23 @@ See 'checkEquation',
 -}
 checkFunctions ::
     MonadLog m =>
-    MonadSimplify m =>
+    MonadMask m =>
+    MonadProf m =>
+    MonadIO m =>
+    MonadSMT m =>
     -- | The main module
     VerifiedModule StepperAttributes ->
     m ()
-checkFunctions verifiedModule = do
+checkFunctions verifiedModule = evalSimplifier verifiedModule' $ do
+    -- let equationMap = extractEquations verifiedModule'
+    equationMap <-
+        extractEquations verifiedModule'
+            & (Map.map . map . Equation.mapVariables) (pure mkEquationVariable)
+            & simplifyExtractedEquations
```

Comment:
Is `simplifyExtractedEquations` necessary or does it just obfuscate the error message?

Reply:
Was this being called before your changes?

I think that calling this function is the actual issue here: this will apply unification, pushing the argument back into the term.

Reply:
@ana-pantilie  It was not being called before and the error message was unsimplified. But adding it did not resolve the error either, it just simplified the error message. So I don't think it is the source of the error. I will probably remove it because I think seeing the unsimplified equation will make the error easier to track down for a user. Thoughts?